index.js

```diff
@@ -146,55 +146,59 @@ function requestAsEventEmitter(opts) {
 			let downloaded = 0;
 
 			setImmediate(() => {
-				const progressStream = new Transform({
-					transform(chunk, encoding, callback) {
-						downloaded += chunk.length;
+				try {
```

Comment:
Can you move the contents of the `try` into a function and call it from the `try`, so we don't add more nesting.